cmd/pulumi.go

```diff
@@ -155,6 +175,170 @@ func NewPulumiCmd() *cobra.Command {
 	return cmd
 }
 
+// checkForUpdate checks to see if the CLI needs to be updated, and if so emits a warning, as well as information
+// as to how it can be upgraded.
+func checkForUpdate() {
+	curVer, err := semver.ParseTolerant(version.Version)
+	if err != nil {
+		glog.V(3).Infof("error parsing current version: %s", err)
+	}
+
+	// We don't care about warning for you to update if you have installed a developer version
+	if isDevVersion(curVer) {
+		return
+	}
+
+	latestVer, oldestAllowedVer, err := getCLIVersionInfo()
+	if err != nil {
+		glog.V(3).Infof("error fetching latest version information: %s", err)
+	}
+
+	if oldestAllowedVer.GT(curVer) {
+		cmdutil.Diag().Warningf(diag.RawMessage("", getUpgradeMessage(latestVer, curVer)))
+	}
+}
+
+// getCLIVersionInfo returns information about the latest version of the CLI and the oldest version that should be
+// allowed without warning. It caches data from the server for a day.
+func getCLIVersionInfo() (semver.Version, semver.Version, error) {
+	latest, oldest, err := getCachedVersionInfo()
```

Comment:
Nit: A comment along the lines of "ignore errors getting cached version info, since it will return an error if the cache is too old." would be helpful.